### PR TITLE
Feat/optim.scss

### LINKED ISSUE #788: optim build time
a lot of time in the build of a angular spa seems to be in the scss transpilation and that may be due to the fact that some componets, in their scss, import the whole theming

there might be some thruth in that, and if so we need to devise a way to optimize this

theoriticaly a component doesn't need the whole theming map, just palettes, colors, sizes and commons

### PR DESCRIPTION
**Pull request by lucienbertin**
fixes #788 

- the components themes (and overrides) are only imported just before importing the rest of the components style (see [components.overridable](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-d35b8114fc99740cdb39ce5288ca1b95R1-R4) and [components](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-ee33917c7baa09e62c936caacf428ae7R1-R3) and [theming](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-dfce7845bd8ca415cd92ecb3f18c297aL13) that no longer imports the components themes)
- added missing theme varables in ng components definition, because for example the definition of a select input cannot use [textgield padding](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-bd28bc5b6bd826013ab8b2057fb439a3L205-L206) or [framed padding](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-bd28bc5b6bd826013ab8b2057fb439a3L128-L133)

-----

just empiric results but when i run `ng build sandbox` i go from `69 seconds` to `29 seconds`

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
Travis automatic deployment: https://830-pr-lucca-front-LuccaSA.surge.sh

File: packages/ng/libraries/core/src/style/theme/_select.theme.scss [hunk 1]
```diff
@@ -3,6 +3,21 @@ $select: (
 	chip: (
 		background: _theme("commons.background.dark"),
 		text: _color("text.default")
+	),
+	input: (
+		padding-horizontal: .5rem,
+		padding-vertical: .45rem,
```
Comment (sanlucca) [inline packages/ng/libraries/core/src/style/theme/_select.theme.scss]: Dans ce cas, on devrait faire référence à la valeur de thème de textfield ici. Sinon, changer un textfield demanderait de repasser partout. Hors finalement, ce n'est qu'un textfield comme un autre.

Comment (lucienbertin) [reply packages/ng/libraries/core/src/style/theme/_select.theme.scss]: english

i know, what we could do is having textfield theme and select theme both reference a theme variable in `commons`

**Comment by lucienbertin** [comment]
| project | command | before | after | % |
| ----- | ----- | ---:| ---: | ---:|
| lf sandbox | build | 69 | 29 | -55%|
| admin cals | build prod | 85 | 77 |-10% |
| admin cals | serve | 30 | 30 | - |
| tproj | build prod | 425 | 227 | -45% |
| tproj | serve | 182 | 81 | -55% |

File: packages/ng/libraries/core/src/style/theme/_select.theme.scss (hunk 1 above)
Comment (lucienbertin) [reply packages/ng/libraries/core/src/style/theme/_select.theme.scss]: also, a `lu-select` should only have the style of a textfield when incorporrated in a `.textfield`

```html
<div class="textfield">
  <lu-select class="textfield-input">...</lu-select>
  <label class="textfield-label">label</label>
</div>
```

as a result i dont know why having the `_component('textfield.padding.right')` [here](https://github.com/LuccaSA/lucca-front/pull/830/files#diff-bd28bc5b6bd826013ab8b2057fb439a3L17-L18) is mandatory

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-830

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-830

**Comment by lucienbertin** [comment]
@sanlucca your remarks can be integrated in a follow up PR

**Pull request merged**